Curve-finance-suite/Registry-Add-Pool/src/agent.spec.ts

```diff
@@ -0,0 +1,92 @@
+import {
+  Finding,
+  FindingSeverity,
+  FindingType,
+  HandleTransaction,
+  TransactionEvent,
+} from 'forta-agent';
+
+import { createAddress, TestTransactionEvent } from 'forta-agent-tools';
+
+import { provideHandleTransaction, ADD_POOL_SIGNATURE, R_IFACE } from './agent';
+
+const TARGET: string = createAddress('0xdead');
+const ALERT_ID: string = 'registry-add-pool-test';
+const RECEIVERS: string[] = [];
+const SENDERS: string[] = [];
+
+const createFinding = ( pool: string) => Finding.fromObject({
+  name: 'Curve Registry contract called',
+  description: 'Event PoolAdded has been emitted',
+  alertId: ALERT_ID,
+  severity: FindingSeverity.Info,
+  type: FindingType.Info,
+  protocol: 'Curve Finance',
+  metadata: {
+    pool_address: pool,
+  }
+});
+
+describe('Registry-Add-Pool Agent tests suite', () => {
+  const handler: HandleTransaction = provideHandleTransaction(
+    ALERT_ID,
+    TARGET,
+  );
+
+  beforeAll(() => {
+    for(let i = 0; i <= 30; ++i) {
+      RECEIVERS.push(createAddress(`0xa${i}`));
+      SENDERS.push(createAddress(`0xb${i}`));
+    }
+  });
+
+  it('should ignore empty transactions', async () => {
+    const tx: TransactionEvent = new TestTransactionEvent();
+
+    const findings = await handler(tx);
+    expect(findings).toStrictEqual([]);
+  });
+
+  it('should ignore events from other contracts', async () => {
+    // Encode the pool address value
+    let encodedTopicData = R_IFACE.encodeFilterTopics(
+      R_IFACE.getEvent('PoolAdded'),
+      ['0xab5801a7d398351b8be11c439e05c5b3259aec9b'],
+    );
+    const encodedPoolAddress: string = encodedTopicData[1] as string;
```

Comment:
- we can use `encodeParameter` from` forta-agent-tools` here, ie:
` encodedPoolAddress = encodeParameter('address','0xab5801a7d398351b8be11c439e05c5b3259aec9b');`
- this part can be moved outside to avoid repeating the same declaration inside each test case